File: ui/src/commands/ranking.ts
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

import { invoke } from '@tauri-apps/api/core';

export function getBotRank(id: BotId) {
  return getRank({ kind: 'bot', id });
}

export function getPlayerRank(id: PlayerId) {
  return getRank({ kind: 'player', id });
}

export function getPrecursorRank(id: PrecursorId) {
  return getRank({ kind: 'precursor', id });
}

export function getRank(ruler: Ruler) {
  return invoke<Option<RankingEntry>>('get_rank', { ruler });
}

export function getRanking() {
  return invoke<Ranking>('get_ranking');
}

File: crates/nil/src/command/ranking.rs
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

use crate::error::Result;
use crate::manager::ManagerExt;
use nil_core::ranking::{Ranking, RankingEntry};
use nil_core::ruler::Ruler;
use tauri::AppHandle;

#[tauri::command]
pub async fn get_rank(app: AppHandle, ruler: Ruler) -> Result<Option<RankingEntry>> {
  app
    .client(async |cl| cl.get_rank(ruler).await)
    .await?
    .map_err(Into::into)
}

#[tauri::command]
pub async fn get_ranking(app: AppHandle) -> Result<Ranking> {
  app
    .client(async |cl| cl.get_ranking().await)
    .await?
    .map_err(Into::into)
}
